src/api/cvc4cpp.cpp

```diff
@@ -2073,6 +2073,365 @@ CVC4::Expr Term::getExpr(void) const
+  const Rational& r = getRational(*d_node);
+  return std::make_pair(r.getNumerator().getSignedInt(),
+                        r.getDenominator().getUnsignedInt());
+}
+bool Term::isReal64() const
+{
+  return isRealString() && checkReal64Bounds(getRational(*d_node));
+}
+std::pair<std::int64_t, std::uint64_t> Term::getReal64() const
+{
+  CVC4_API_CHECK(isReal64())
+      << "Term should be a Real64 when calling getReal64()";
+  const Rational& r = getRational(*d_node);
+  return std::make_pair(r.getNumerator().getLong(),
+                        r.getDenominator().getUnsignedLong());
+}
+bool Term::isRealString() const
+{
+  return d_node->getKind() == CVC4::Kind::CONST_RATIONAL;
+}
+std::string Term::getRealString() const
+{
+  CVC4_API_CHECK(isRealString())
+      << "Term should be a Real when calling getRealString()";
+  return getRational(*d_node).toString();
+}
+
+bool Term::isPi() const { return d_node->getKind() == CVC4::Kind::PI; }
+
+bool Term::isConstArray() const
+{
+  return d_node->getKind() == CVC4::Kind::STORE_ALL;
+}
+Term Term::getConstArray() const
+{
+  CVC4_API_CHECK(isConstArray())
+      << "Term should be a ConstArray when calling getConstArray()";
+  const auto& ar = d_node->getConst<ArrayStoreAll>();
+  return Term(d_solver, ar.getValue());
+}
+
+bool Term::isEmptyBag() const
+{
+  return d_node->getKind() == CVC4::Kind::EMPTYBAG;
+}
+
+bool Term::isBoolean() const
+{
+  return d_node->getKind() == CVC4::Kind::CONST_BOOLEAN;
+}
+bool Term::isFalse() const
+{
+  return isBoolean() && !d_node->getConst<bool>();
+}
+bool Term::isTrue() const
+{
+  return isBoolean() && d_node->getConst<bool>();
+}
+bool Term::getBoolean() const
+{
+  CVC4_API_CHECK(isBoolean())
+      << "Term should be a Boolean when calling getBoolean()";
+  return d_node->getConst<bool>();
+}
+
+bool Term::isBitVector() const
+{
+  return d_node->getKind() == CVC4::Kind::CONST_BITVECTOR;
+}
+std::string Term::getBitVector(std::uint32_t base) const
+{
+  CVC4_API_CHECK(isBitVector())
+      << "Term should be a BitVector when calling getBitVector()";
+  return d_node->getConst<BitVector>().toString(base);
+}
+
+bool Term::isAbstractValue() const
+{
+  return d_node->getKind() == CVC4::Kind::ABSTRACT_VALUE;
+}
+std::string Term::getAbstractValue() const
+{
+  CVC4_API_CHECK(isAbstractValue())
+      << "Term should be a AbstractValue when calling "
+         "getAbstractValue()";
+  return d_node->getConst<AbstractValue>().getIndex().toString();
+}
+
+bool Term::isTuple() const
+{
+  return d_node->getKind() == CVC4::Kind::APPLY_CONSTRUCTOR;
+}
+std::vector<Term> Term::getTuple() const
+{
+  CVC4_API_CHECK(isTuple()) << "Term should be a Tuple when calling getTuple()";
+  std::vector<Term> res;
+  for (std::size_t i = 1, n = d_node->getNumChildren(); i < n; ++i)
+  {
+    res.emplace_back(Term(d_solver, (*d_node)[i]));
+  }
+  return res;
+}
+
+bool Term::isPosZero() const
+{
+  if (isFloatingPoint())
```

Comment:
API functions should not call API functions since we will get issues in the future when we implement API tracing.

Reply:
I think this is fixed for all methods from this PR. There are still cases of this pattern, for example for `Sort`.